### PR TITLE
Delete conda builds

### LINKED ISSUE #1365: [Release] Delete Conda Nightly builds
### 🐛 Describe the bug

PyTorch Core is deprecating Nightly Conda builds. see https://github.com/pytorch/pytorch/issues/138506 

We need to clean ours up too, preferably before core deletes on Nov. 18 so we don't have broken builds.

cc @atalman 

### Versions

nightly

### PR DESCRIPTION
**Pull request by ramanishsingh**
PyTorch Core is [deprecating](https://github.com/pytorch/pytorch/issues/138506) Nightly Conda builds. 
This PR aims to delete conda builds
Fixes #1365

### CODE REVIEW

File: .github/workflows/pull_release.yml [hunk 1]
```diff
@@ -7,6 +7,10 @@ on:
       - .github/workflows/pull_release.yml
       - .github/workflows/_build_test_upload.yml
 
+permissions:
+  contents: write
+  id-token: write
+
```
Comment (andrewkho) [inline .github/workflows/pull_release.yml:13]: Do we need this change?

Comment (ramanishsingh) [reply .github/workflows/pull_release.yml:13]: I was getting this error 

`The workflow is not valid. .github/workflows/pull_release.yml (Line: 11, Col: 3): Error calling workflow 'pytorch/data/.github/workflows/_build_test_upload.yml@45dc471d9a16eb4cf482f38b9167f99a13cd55d4'. The workflow is requesting 'contents: write, id-token: write', but is only allowed 'contents: read, id-token: none'.`

https://github.com/pytorch/data/actions/runs/11864790565

After I added these lines it worked. I can try to again remove these and see what happens now.

File: packaging/post_build_script_linux.sh [hunk 2]
```diff
@@ -7,8 +7,8 @@ for pkg in dist/torchdata*.whl; do
     echo "PkgInfo of $pkg:"
     pkginfo $pkg
 
-    auditwheel repair $pkg --plat manylinux2014_x86_64 -w wheelhouse
```
Comment (atalman) [inline packaging/post_build_script_linux.sh:10]: Looks like we are mixing manylinux here and conda removal

File: .github/workflows/pull_release.yml (hunk 1 above)
Comment (atalman) [reply .github/workflows/pull_release.yml:13]: yes this maybe needed

File: packaging/post_build_script_linux.sh (hunk 2 above)
Comment (ramanishsingh) [reply packaging/post_build_script_linux.sh]: Thanks, updated (reverted).

**Review by atalman — APPROVED** [review]
lgtm

**Pull request merged**